rasa/core/processor.py

```diff
@@ -142,6 +142,33 @@ def __init__(
             "tracker": tracker.current_state(EventVerbosity.AFTER_RESTART),
         }
 
+    async def _run_session_start_sequence(
+        self, tracker: DialogueStateTracker, output_channel: OutputChannel
+    ) -> None:
+        """Run `ACTION_SESSION_START_NAME` and any follow-up actions triggered by it.
+
+        Args:
+            tracker: Tracker to run the actions on.
+            output_channel: Output channel for potential utterances in a custom
+                `ActionSessionStart`.
+        """
+
+        action = self._get_action(ACTION_SESSION_START_NAME)
+
+        while action:
```

Comment:
I feel it's weird if we haven another (dummy) prediction loop in the code. Imo the condition for `should_predict_another_action` is wrong and should be `should_predict_another_action(tracker.latest_action_name)`. What do you think?

Reply:
agreed - updated 👍